js/app/modules/filter/currentSet.js

```diff
@@ -48,4 +50,26 @@ const CurrentSet = new Module.Class({
 
         return belongs_to_set && !belongs_to_subset;
     },
+
+    // Filter override
+    // This Filter can modify the Xapian query, but it can't express its whole
+    // filtering logic in the Xapian query, so we still need to request more
+    // models in case some returned models must be dropped.
+    can_modify_xapian_query: function () {
```

Comment:
@ptomato couldn't this be implemented by just using the SetMap data to..

tags_match_all: [current_set_tag]
AND
excluded_tags:[list_of_subsets_of_current_set].

?